src/hooks/queries/productList/useProductListQuery.ts

```diff
@@ -0,0 +1,40 @@
+import { useInfiniteQuery } from '@tanstack/react-query';
+
+import { PRODUCT_KEYS } from './queryKeys';
+import useNetworkStatus from '../../useNetworkStatus';
+
+import { fetchProductList } from '@/api/product';
+import { PAGE_SIZE } from '@/constants/config';
+import { ProductCategory, SortValue } from '@/types/product';
+
+interface FetchProductListProps {
+  category?: ProductCategory;
+  sortOptions?: SortValue;
+}
+
+const useProductListQuery = ({ category, sortOptions }: FetchProductListProps) => {
+  useNetworkStatus();
+
+  return useInfiniteQuery({
+    queryKey: PRODUCT_KEYS.filter([category, sortOptions]),
+    queryFn: ({ pageParam }) => {
+      const size = pageParam === 0 ? PAGE_SIZE.firstItemSize : PAGE_SIZE.nextItemSize;
+
+      return fetchProductList({ category, size, page: pageParam, sortOptions });
+    },
+    staleTime: 300 * 1000,
```

Comment:
5분을 표현하려고 한거 같은데 아래처럼 적으면 직관적으로 이해하기 좋을거 같아요
```suggestion
    staleTime: 5 * 60 * 1_000,
```

Reply:
staleTime을 5분으로 잡은 이유가 있나요?

Reply:
60초 단위로 작성하는 게 훨씬 직관적인 것 같아요!

5분의 큰 의미는 없고 캐싱은 되지만 다시 refetch도 되어야하지 않을까 생각해서 5분을 잡게 되었습니다. 하지만 굳이 refetch를 할 필요가 없다고 생각이 들어 staleTime을 Infinity로 설정하고 mutation이 발생할 경우 invalidate하는 구조로 수정하였습니다!

Reply:
>이커머스 서비스를 개발하게 된다면 상품이 추가 및 품절될 수도 있으니까 Infinity로 설정하는 것이 적절한가?

부적절하다고 생각합니다 Infinity로 설정하게 되면 서버와 클라이언트의 데이터 정합성을 보장하기 어렵기 때문인데요. 이전 리뷰에 남겼던 비마트 포켓몬 빵 사례를 한번 생각해보시면 좋을거 같아요!

>서버에서 한번더 validation을 거치겠지만 staleTime을 어떤 기준으로 설정하시는지 유조의 생각도 궁금합니다!

이것도 이전 리뷰에 말씀드린 낙관적 업데이트와 약간 비슷한 맥락을 가져가는데요. 클라이언트와 서버와 데이터 정합성이 얼마나 중요한지, 해당 요청이 서버에 부하를 유발할 수 있는지 등을 고려하여 결정하게 됩니다.   ㅎㅎ